File: slnSopimusrekisteriLuokat/Sopimusrekisteri.BLL_CF/Maksu.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Sopimusrekisteri.BLL_CF
{
  public class Maksu
  {
		#region Propertyt

		public int Id {get; private set;}
		public int? MaksuaineistoId {get; set;}
		public int? KorvauslaskelmaId {get; set;}
		public DateTime? Maksupaiva {get; set;}
		public decimal? Summa {get; set;}
		public decimal? Vero {get; set;}
		public string Tilinumero {get; set;}
		public string Bic {get; set;}
		public string Viite {get; set;}
		public string Viesti {get; set;}
		public int? IndeksiKuukausiId {get; set;}
		public int? Indeksi {get; set;}
		public int? KirjanpidonTiliId {get; set;}
		public string Kustannuspaikka {get; set;}
		public int? MaksuStatusId {get; set;}
		public string LaskunNumero {get; set;}
		public string EraTunniste {get; set;}
		public DateTime? AjoPvm {get; set;}
		public int? Vuosi {get; set;}
		public string Info {get; set;}
		public DateTime Luotu {get; set;}
		public string Luoja {get; set;}
		public int? SaajaId {get; set;}
    public string SaajaNimi {get; set;}
		public bool OnIndeksi {get; set;}
		public int? IndeksityyppiId {get; set;}
		public int? MaksuIndeksi {get; set;}
		public decimal? AlvOsuus {get; set;}
		public int? JuridinenYhtioId {get; set;}
		public string MaksajanNimi {get; set;}
		public string MaksajanTilinro {get; set;}
		public string MaksajanBicKoodi {get; set;}
		public int? SopimusId {get; set;}
		public string KirjanpidonTunniste {get; set;}
		public string Palvelutunnus {get; set;}
		public bool Passivoitu {get; set;}
		public decimal? SummaIlmanAlv {get; set;}
		public int? IndeksiVuosi {get; set;}
    public DateTime? IfsMaksupvm { get; set; }
    public string IfsLaskunro { get; set; }

    public decimal? Alv { get; set; }

		#endregion

		#region Entiteettiviittaukset

		public virtual Maksuaineisto Maksuaineisto {get; set;}
		public virtual Korvauslaskelma Korvauslaskelma {get; set;}
		public virtual Kuukausi IndeksiKuukausi {get; set;}
		public virtual Kirjanpidontili KirjanpidonTili {get; set;}
		public virtual MaksuStatus MaksuStatus {get; set;}
		public virtual Taho Saaja {get; set;}
		public virtual Indeksityyppi Indeksityyppi {get; set;}
		public virtual Taho JuridinenYhtio {get; set;}
		public virtual Sopimus Sopimus {get; set;}

    public virtual ICollection<MaksuTiliointi> Tilioinnit { get; set; }

		#endregion

		#region Metodit

		public override bool Equals(object obj)
		{
			var tmp = obj as Maksu;
			if (tmp == null) return false;
			return tmp.Id == Id;
		}

		public override int GetHashCode()
		{
			return Id.GetHashCode();
		}

		public override string ToString()
		{
			return Id.ToString();
		}

		#endregion
  }
}

File: appSopimusrekisteri.DTO/Maksuaineistot/Maksu.vb
Public Class Maksu

  Public Property Id As Integer?
  Public Property MaksuEraTunniste As Integer?
  Public Property KorvauslaskelmaId As Integer?
  Public Property Ajopvm As Date?
  Public Property MaksuStatusId As Integer?
  Public Property Maksupvm As Date?
  Public Property Summa As Decimal?
  Public Property Alv As Decimal?
  Public Property Viite As String
  Public Property Viesti As String
  Public Property Vuosi As Integer?
  Public Property Tilinumero As String
  Public Property BicKoodi As String
  Public Property SaajaId As Integer?
  Public Property Saaja As String
  Public Property Luoja As String
  Public Property Luotu As Date?
  Public Property OnIndeksi As Boolean = False
  Public Property Indeksi As Integer?
  Public Property MaksuIndeksi As Integer?
  Public Property IndeksikuukausiId As Integer?
  Public Property Indeksikuukausi As String
  Public Property IndeksityyppiId As Integer?
  Public Property Indeksityyppi As String
  Public Property Info As String
  Public Property AlvOsuus As Decimal?
  Public Property JuridinenYhtioId As Integer?
  Public Property MaksajanNimi As String
  Public Property MaksajanTilinro As String
  Public Property MaksajanBicKoodi As String
  Public Property SopimusId As Integer?
  Public Property SummaIlmanAlv As Decimal?
  Public Property Indeksivuosi As Integer?

  Public Property Palvelutunnus As String
  Public Property Kirjanpidontunniste As String
  Public Property KirjanpidonProjektitunniste As String
  Public Property JuridinenYhtio As Taho

  Public Property Korvauslaskelma As Korvauslaskelma
  Public Property Tiliointi As MaksunTiliointi()

End Class
